src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -342,13 +343,31 @@ public static Translation2d[] getModuleTranslations() {
     };
   }
 
-  public Rotation2d getRotationToTranslation(Translation2d translation) {
-
+  public Rotation2d getRotationToTranslation(Pose2d translation) {
     double angle = Math.atan2(translation.getY() - pose.getY(), translation.getX() - pose.getX());
     return Rotation2d.fromRadians(angle);
   }
 
-  public Command pointTowardsTranslation(DoubleSupplier x, DoubleSupplier y) {
+  @AutoLogOutput(key = "Autoaim/Target")
+  public Pose2d getVirtualTarget() {
+
+    Pose2d target = FieldConstants.getSpeaker();
+
+    double distance =
```

Comment:
i believe you can get this with methods on pose2d instead of by hand, which is a little cleaner. try `minus` and `norm`